benchmarks/variable_rate.jl

```diff
@@ -0,0 +1,195 @@
+# This file is not directly included in a test case, but is used to
+# benchmark and compare GillespieIntegCallback and NextReactionODE
+using DiffEqBase, JumpProcesses, OrdinaryDiffEq, StochasticDiffEq, Test
+using Random, LinearSolve
+using StableRNGs
+rng = StableRNG(12345)
+
+rate = (u, p, t) -> u[1]
+affect! = (integrator) -> (integrator.u[1] = integrator.u[1] / 2)
+jump = VariableRateJump(rate, affect!, interp_points = 1000)
+jump2 = deepcopy(jump)
+
+f = function (du, u, p, t)
+    du[1] = u[1]
+end
+
+prob = ODEProblem(f, [0.2], (0.0, 10.0))
+
+jump_prob = JumpProblem(prob, Direct(), jump, jump2; variablerate_aggregator = NextReactionODE(), rng = rng)
+jump_prob_gill = JumpProblem(prob, Direct(),  jump, jump2; variablerate_aggregator = GillespieIntegCallback(), rng=rng)
+
+integrator = init(jump_prob, Tsit5())
+integrator = init(jump_prob_gill, Tsit5())
+
+time_next = @elapsed solve(jump_prob, Tsit5())
+time_gill = @elapsed solve(jump_prob_gill, Tsit5())
+
+println("Time taken for GillespieIntegCallback $time_gill")
+println("Time taken for NextReactionODE $time_next")
+
+
+time_next = @elapsed solve(jump_prob,  Rosenbrock23(autodiff = false))
+time_gill = @elapsed solve(jump_prob_gill,  Rosenbrock23(autodiff = false))
+
+println("Time taken for GillespieIntegCallback $time_gill")
+println("Time taken for NextReactionODE $time_next")
+
+time_next = @elapsed solve(jump_prob,  Rosenbrock23())
+time_gill = @elapsed solve(jump_prob_gill,  Rosenbrock23())
+
+println("Time taken for GillespieIntegCallback $time_gill")
+println("Time taken for NextReactionODE $time_next")
+
+
+
+g = function (du, u, p, t)
+    du[1] = u[1]
+end
+
+prob = SDEProblem(f, g, [0.2], (0.0, 10.0))
+
+jump_prob = JumpProblem(prob, Direct(), jump, jump2; variablerate_aggregator = NextReactionODE(), rng = rng)
+jump_prob_gill = JumpProblem(prob, Direct(),  jump, jump2; variablerate_aggregator = GillespieIntegCallback(), rng=rng)
+
+time_next = @elapsed solve(jump_prob,  SRIW1())
+time_gill = @elapsed solve(jump_prob_gill,  SRIW1())
```

Comment:
Single small solves of a stochastic process don't make too much sense. Can you instead time ensembles of say 1000 solves?